sympy/core/basic.py

```diff
@@ -1034,21 +1035,23 @@ def subs(self, *args, **kwargs):
         from .containers import Dict
         from .symbol import Dummy, Symbol
         from .numbers import _illegal
+        from .relational import Equality
 
         unordered = False
         if len(args) == 1:
-
             sequence = args[0]
             if isinstance(sequence, set):
                 unordered = True
             elif isinstance(sequence, (Dict, Mapping)):
                 unordered = True
                 sequence = sequence.items()
+            elif isinstance(sequence, Equality):
+                sequence = sequence.args
```

Comment:
```suggestion
```

Reply:
handling of Eq is being done in the loop below.

Please add a test like `assert x.subs(Eq(x, y)) == y` and `assert (x + y).subs([Eq(y, 1), (x, y)]) == 1 + y`.

Please add yourself as an author (like in #26403).

Reply:
This still needs to be here to allow a single equality arg to be passed into the function.